### PR TITLE
[TEST] wait for yellow after setup doc tests

### PR DESCRIPTION
**Pull request by brwe**
We have many places in the doc where we expect and index to be
yellow before we execute a query. Therefore we have to
always wait for yellow after setup.

Related test failure (I think): https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/606/consoleText

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTask.groovy [hunk 1]
```diff
@@ -170,6 +178,7 @@ public class RestTestsFromSnippetsTask extends SnippetsTask {
             current.println('---')
             current.println("setup:")
             body(setup)
+            body(waitForYellow)
```
Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/doc/RestTestsFromSnippetsTask.groovy]: I'd probably just `current.println` the wait_for_yellow stuff. Like, you can even use normal rest-yaml syntax and everything in here.

**Comment by nik9000** [comment]
I'm still hoping for https://github.com/elastic/elasticsearch/issues/9126 but in the mean time this should help. It won't get everything because we have cases where we use make an index and use it in the same snippet, but this'll still help.

**Comment by brwe** [comment]
@nik9000 addressed your comment. Thank you!

**Comment by nik9000** [comment]
LGTM

**Pull request merged**